ssc_utils/filter_generator.py

```diff
@@ -0,0 +1,288 @@
+class filter_generator(object):
+    """
+    Contains a set of functions that generates the SQL CTEs that go through CUPED calculations. 
+    Should always be the first CTE in the final SQL string.     
+    """
+    
+    def attribute_conditions_choices(self):
+        attribute_conditions = [
+            '=',
+            '<>',
+            'IN',
+            'IS',
+            'IS NOT'
+        ]
+        return attribute_conditions 
+    
+    def metric_conditions_choices(self):
+        metric_conditions =  [
+            '>',
+            '<',
+            '>=',
+            '<=',
+            'BETWEEN'
+        ] + self.attribute_conditions_choices()
+        return metric_conditions 
+
+    def filter_attributes_choices(self):
+        """
+        List of attributes available for filtering (from all_metric_hourly)
+        """ 
+        filter_attributes = [
+                'no filters',
+                'user_id',
+                'device_id',
+                'platform',
+                'platform_type',
+                'country',
+                'region',
+                'city',
+                'dma',
+                'os',
+                'os_version',
+                'manufacturer',
+                'app_mode',
+                'app_version',
+                'device_language',
+                'content_id',
+                'program_id',
+                'content_type',
+                'tvt_sec' # note: here tvt_sec is treated as an attribute rather than a cumulative metric
+            ]
+        return filter_attributes 
+
+    def filter_metrics_choices(self):
+        """
+        List of metrics available for filtering (from all_metric_hourly)
+        """         
+        filter_metrics = [
+            'no filters',
+            'tvt_sec',
+            'movie_non_autoplay_tvt_sec',
+            'series_non_autoplay_tvt_sec',
+            'autoplay_tvt_sec',
+            'non_autoplay_tvt_sec',
+            'series_tvt_sec',
+            'movie_tvt_sec',
+            'series_autoplay_tvt_sec',
```

Comment:
would it be a good idea to pull all the column names from the schema directly, then filter by having the string "sec" or "count"?

this could be done by:
```
select column_name
from information_schema.columns
where table_catalog = 'tubidb'
and table_schema = 'tubidw'
and table_name = 'all_metric_hourly'
```

Reply:
that's a great idea! will be much easier to maintain :)

Reply:
I don't think TDR has access to information_schema :(

Reply:
replaced with `tdr.get_catalog().tubidw.all_metric_hourly.columns`